src/extension/extensionServer.ts

```diff
@@ -0,0 +1,141 @@
+
+        let deferred = Q.defer<void>();
+
+        let launchCallback = (error: any) => {
+            Log.logMessage("Server started." + (error ? error : ""));
+            if (error) {
+                deferred.reject(error);
+            } else {
+                deferred.resolve(null);
+            }
+        };
+
+        this.serverInstance = net.createServer(this.handleSocket.bind(this));
+        this.serverInstance.on("error", this.recoverServer.bind(this));
+        this.serverInstance.listen(em.getPipePath(), launchCallback);
+
+        return deferred.promise;
+    }
+
+    /**
+     * Stops the server.
+     */
+    public dispose(): void {
+        if (this.serverInstance) {
+            this.serverInstance.close();
+            this.serverInstance = null;
+        }
+    }
+
+    /**
+     * Message handler for START_PACKAGER.
+     */
+    private startPackager(): Q.Promise<any> {
+        return this.reactNativePackager.start(this.outputChannel);
+    }
+
+    /**
+     * Message handler for STOP_PACKAGER.
+     */
+    private stopPackager(): Q.Promise<any> {
+        return this.reactNativePackager.stop(this.outputChannel);
+    }
+
+    /**
+     * Message handler for PREWARM_BUNDLE_CACHE.
+     */
+    private prewarmBundleCache(platform: string): Q.Promise<any> {
+        return this.reactNativePackager.prewarmBundleCache(platform);
+    }
+
+    /**
+     * Extension message handler.
+     */
+    private handleExtensionMessage(messageWithArgs: em.MessageWithArguments): Q.Promise<any> {
+        let handler = this.messageHandlerDictionary[messageWithArgs.message];
+        if (handler) {
+            Log.logMessage("Handling message: " + em.ExtensionMessage[messageWithArgs.message]);
+            return handler.apply(this, messageWithArgs.args);
+        } else {
+            return Q.reject("Invalid message: " + messageWithArgs.message);
+        }
+    }
+
+    /**
+     * Handles connections to the server.
+     */
+    private handleSocket(socket: net.Socket): void {
+        let dataCallback = (data: any) => {
+            try {
+                let messageWithArgs: em.MessageWithArguments = JSON.parse(data);
+                this.handleExtensionMessage(messageWithArgs)
+                    .then(result => {
+                        socket.end(JSON.stringify(result));
+                    })
+                    .catch(() => { socket.end(); })
+                    .done();
+            } catch (e) {
+                socket.end();
+            }
+        };
+
+        socket.on("data", dataCallback);
+    };
+
+    /**
+     * Recovers the server in case the named socket we use already exists, but no other instance of VSCode is active.
+     */
+    private recoverServer(error: any): void {
+        let errorHandler = (e: any) => {
+            /* The named socket is not used. */
+            if (e.code === "ECONNREFUSED") {
+                new FileSystem().removePathRecursivelyAsync(em.getPipePath())
```

Comment:
That is what `removePathRecursivelyAsync` does internally - removes a directory or a file, if they exist.